warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java

```diff
@@ -476,6 +477,33 @@ public boolean removeAll(Collection<?> c) {
         return modified;
     }
     
+    @Override
+    public boolean removeIf(Predicate<? super E> filter) {
+        boolean removed = false;
+        for (SortedSet<E> subSet : set.getSets()) {
+            FileSortedSet<E> fileSet = (FileSortedSet<E>) subSet;
+            if (fileSet.isPersisted()) {
+                try {
+                    fileSet.load();
+                    if (fileSet.removeIf(filter)) {
+                        removed = true;
+                    }
+                    fileSet.persist();
```

Comment:
only persist if removed?

Reply:
nope, I wanted to put it back into a persisted state if it was previously persisted.

Reply:
But I get your point, if nothing changed then why rewrite the files.  I will create a separate ticket to add a mechanism to flip from loaded to persisted using the existing underlying files.

Reply:
see #949